### PR TITLE
Ixe 8279

### PR DESCRIPTION
**Pull request by bethleha**
@paambaati Please review this.

### CODE REVIEW

File: src/util.js [hunk 1]
```diff
@@ -27,10 +27,15 @@ export function convertToQueryParams(o){
       case 'brandId':
       case 'categoryId':
       case 'facetBy':
-        let x = _.map(value, function(value){
-          return attr + '=' + value;
-        });
-        return x.join('&');
+        if (_.isArray(value)) {
+          let x = _.map(value, function(value){
```
Comment (paambaati) [inline src/util.js:31]: Let's not use lodash for trivial native methods. This was already pointed out [here](https://github.com/indix/indix-api-nodejs/pull/11#discussion_r107118419) and [here](https://github.com/indix/indix-api-nodejs/pull/11#discussion_r107118663). I'd recommend going over the review comments once again on #11 carefully. Also please avoid the use of single-character variable names like `x`; variable names should be self-explanatory.

File: test/product-search-test.js [hunk 2]
```diff
@@ -59,6 +59,98 @@ describe('Product Search', function() {
 
   });
 
+  it('summay should return appropriate products from store/s or brand/s 1', function(done){
```
Comment (paambaati) [inline test/product-search-test.js]: Typo in summary. This was also pointed out earlier on #11 [here](https://github.com/indix/indix-api-nodejs/pull/11#discussion_r107125189).

File: src/util.js (hunk 1 above)
Comment (bethleha) [reply src/util.js:31]: lodash is used in lot of places in this api-client and its usage is not optimized. Will apply this feedback while refactoring this api-client.

eg. _.map which is equivalent of Array.map is consistently used in this api-client.

 We can even refactor this api-client to completely get rid of lodash.

File: test/product-search-test.js (hunk 2 above)
Comment (bethleha) [reply test/product-search-test.js]: corrected in the following commit.

File: src/util.js (hunk 1 above)
Comment (paambaati) [reply src/util.js:31]: > lodash is used in lot of places in this api-client and its usage is not optimzied.

Meaning?

Comment (bethleha) [reply src/util.js:31]: Updated my last comment.

File: test/product-search-test.js [hunk 3]
```diff
@@ -59,6 +59,98 @@ describe('Product Search', function() {
 
   });
 
+  it('summary should return appropriate products from store/s or brand/s 1', function(done){
```
Comment (paambaati) [inline test/product-search-test.js:62]: @bethleha Can we not qualify these test cases with numbers but instead add a more elaborate description?

Comment (bethleha) [reply test/product-search-test.js:62]: @paambaati They are checking the same case with different inputs. Won't the description help to identify that the same case is tested with 2 different inputs?

**Pull request merged**